bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details/SourceDetailsEditor.scala

```diff
@@ -274,43 +258,106 @@ final class SourceDetailsEditor extends GridBagPanel with TelescopePosEditor wit
+        case Some(s: AuxFileSpectrum) => Some(s)
+        case _                        => None
+      }
 
-  // get a list of all available sed files (aux files ending in ".sed")
-  private def sedFiles(programId: SPProgramID): Option[Seq[AuxFileSpectrum]] = {
-    try {
-      import scala.collection.JavaConversions._
-      VcsOtClient.unsafeGetRegistrar.registration(programId).map { peer =>
-        val aux    = new AuxFileClient(OT.getKeyChain, peer.host, peer.port)
-        val files  = aux.listAll(programId)
-        files.
-          filter(AuxFileType.getFileType(_) == AuxFileType.sed).
-          map   (f => AuxFileSpectrum(programId.toString, f.getName)).
-          toSeq
+      // Update the model and make sure that the currently selected value is one of the available entries.
+      // This allows to keep the current selection in case the aux file listing is currently not available, or if the
+      // currently selected aux file has been removed (we don't want to simply replace an invalid selection with
+      // another value without letting the user know). If the selected file has been removed, calling the ITC will
+      // result in a meaningful error message that tells the user that the selected aux file is not available anymore.
+      def setModel(available: Set[AuxFileSpectrum], selected: Option[AuxFileSpectrum]) = {
+        val all1 = selected.fold(available)(s => available + s)               // add current selection to available files if needed
+        val all2 = if (all1.isEmpty) Set(AuxFileSpectrum.Undefined) else all1 // if there isn't anything add a "dummy" element
+        userDefinedDetails.peer.setModel(ComboBox.newConstantModel(all2.toSeq.sortBy(_.name))) // combo box model musn't be empty
+        selected.foreach(s => userDefinedDetails.selection.item = s)
       }
-    } catch {
-      case _: Exception => None
+
+      SourceDetailsEditor.this.deafTo(editElements:_*)    // DON'T trigger any events!
+      getAuxFiles(programId) match {
+        case files => setModel(files, selectedAuxFile)
+      }
+      SourceDetailsEditor.this.listenTo(editElements:_*)  // OK, done with updating UI elements
+
     }
-  }
 
+    /** Gets a set of all available SED aux files for the given program ID.
+      * The available aux files are filtered and only the ones ending in ".sed" are returned.
+      * This is called in order to update the available aux files in the selection combo box right the moment
+      * when the combobox is expanded to show the selection, therefore if the result is not returned from
+      * the server in a short time, we simply ignore the result.
+      */
+    private def getAuxFiles(programId: SPProgramID): Set[AuxFileSpectrum] = {
+
+      try {
+        import scala.collection.JavaConversions._
+
+        // block UI for at most 3 seconds, if getting the aux file listing takes longer, the result will be ignored
+        Await.result(Future {
```

Comment:
This is done when the user opens the dropdown menu of the combobox, basically the UI is waiting for the values to be displayed at this moment. If I do this non-blocking, the current values (which could be out of date) will be displayed while I am updating the values in the background. I agree this isn't great, but I couldn't come up with a better place to hook up and get notified that it is necessary to update the combobox model.